packages/slider/index.tsx

```diff
@@ -0,0 +1,380 @@
+  min: number;
+  max: number;
+  step?: number;
+  disabled?: boolean;
+  className?: string;
+  tabIndex?: number;
+  isRtl?: boolean;
+  onValueChange?: (value: number) => void;
+  onValueInput?: (value: number) => void;
+}
+
+interface SliderState {
+  classList: Set<string>;
+  markerValue?: number;
+  markerStyles: React.CSSProperties[];
+  trackStyleProperty: React.CSSProperties;
+  thumbContainerStyleProperty: React.CSSProperties;
+}
+
+type SliderElementNames = 'trackStyleProperty' | 'thumbContainerStyleProperty';
+
+class Slider extends React.Component<SliderProps, SliderState> {
+  get classes() {
+    const {classList} = this.state;
+    const {className, discrete, displayMarkers} = this.props;
+    return classnames('mdc-slider', Array.from(classList), className, {
+      'mdc-slider--discrete': discrete,
+      'mdc-slider--display-markers': discrete && displayMarkers,
+    });
+  }
+
+  get adapter(): MDCSliderAdapter {
+    return {
+      hasClass: (className: string): boolean =>
+        this.classes.split(' ').includes(className),
+      addClass: (className: string): void => {
+        const {classList} = this.state;
+        classList.add(className);
+        this.setState({classList});
+      },
+      removeClass: (className: string): void => {
+        const {classList} = this.state;
+        classList.delete(className);
+        this.setState({classList});
+      },
+      getAttribute: (name: string): string | null => {
+        if (!this.sliderElement.current) {
+          return null;
+        }
+        return this.sliderElement.current.getAttribute(name);
+      },
+      setAttribute: (name: string, value: string): void => {
+        if (!this.sliderElement.current) {
+          return;
+        }
+        this.sliderElement.current.setAttribute(name, value);
+      },
+      removeAttribute: (name: string): void => {
+        if (!this.sliderElement.current) {
+          return;
+        }
+        this.sliderElement.current.removeAttribute(name);
+      },
+      computeBoundingRect: (): ClientRect | DOMRect => {
+        if (!this.sliderElement.current) {
+          // new DOMRect is not IE11 compatible
+          const defaultDOMRect = {
+            bottom: 0,
+            height: 0,
+            left: 0,
+            right: 0,
+            top: 0,
+            width: 0,
+            x: 0,
+            y: 0,
+          };
+          return defaultDOMRect;
+        }
+        return this.sliderElement.current.getBoundingClientRect();
+      },
+      getTabIndex: (): number => this.props.tabIndex || 0,
+      // React handles events differently:
+      registerInteractionHandler: () => undefined,
+      deregisterInteractionHandler: () => undefined,
+      registerThumbContainerInteractionHandler: (
+        evtType: EventType,
+        handler: SpecificEventListener<EventType>
+      ): void => {
+        if (evtType === this.transitionendEvtName) {
+          this.onThumbEnd = handler as any;
```

Comment:
can you change `any` to a specific event listener?

Reply:
I'll cast it to `() => void` since that's the type of the function.